### PR TITLE
Adds min primary shard size rollover condition to the ISM rollover ac…

### PR DESCRIPTION
**Pull request by dbbaughe**
…tion

Signed-off-by: Drew Baugher <46505179+dbbaughe@users.noreply.github.com>

*Issue #, if available:*
https://github.com/opensearch-project/index-management/issues/213

*Description of changes:*
* Adds min_primary_shard_size as a rollover condition to ISM rollover action.
* It will evaluate to true when there is a single primary shard in the index that has a greater size than the condition.
* Kept the prefix as min instead of max to be aligned with the other conditions we have in rollover, although it is odd that our conditions are differently named than the native rollover API conditions.

*CheckList:*
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbbaughe** [comment]
TODO: Open a PR on dashboards plugin to add this new option to the UI
This isn't blocking to get this PR in though for now

**Review by qreshi — DISMISSED** [review]

File: src/test/kotlin/org/opensearch/indexmanagement/IndexManagementRestTestCase.kt [hunk 1]
```diff
@@ -106,11 +108,20 @@ abstract class IndexManagementRestTestCase : ODFERestTestCase() {
...
-    protected fun insertSampleData(index: String, docCount: Int, delay: Long = 0, jsonString: String = "{ \"test_field\": \"test_value\" }") {
+    protected fun insertSampleData(index: String, docCount: Int, delay: Long = 0, jsonString: String = "{ \"test_field\": \"test_value\" }", routing: String? = null) {
+        var endpoint = "/$index/_doc/?refresh=true"
+        if (routing != null) endpoint += "&routing=$routing"
         for (i in 1..docCount) {
-            val request = Request("POST", "/$index/_doc/?refresh=true")
+            val request = Request("POST", endpoint)
             request.setJsonEntity(jsonString)
-            client().performRequest(request)
+            client().performRequestAsync(
```
Comment (downsrob) [inline src/test/kotlin/org/opensearch/indexmanagement/IndexManagementRestTestCase.kt:117]: Do you see this as potentially causing flakiness in the future? It seems to me like the test could continue on before all of the sample data is available.

Comment (dbbaughe) [reply src/test/kotlin/org/opensearch/indexmanagement/IndexManagementRestTestCase.kt]: Could definitely happen, was leftover when I was previously increasing shard size by adding more and more documents and this test function was taking a long time. Changed to increasing the payload size instead which doesn't have the issue, reverted.

**Review by downsrob — APPROVED** [review]

**Review by qreshi — APPROVED** [review]

**Comment by sastorsl** [comment]
Should perhaps have updated the docs aswell: [https://opensearch.org/docs/latest/im-plugin/ism/policies/#rollover](https://opensearch.org/docs/latest/im-plugin/ism/policies/#rollover)

**Pull request merged**